bindings/node/lib/clientEncryption.js

```diff
@@ -113,11 +113,12 @@ module.exports = function (modules) {
       // kmsProviders will be parsed by libmongocrypt, must be provided as BSON binary data
       if (options.kmsProviders && !Buffer.isBuffer(options.kmsProviders)) {
         options.kmsProviders = this._bson.serialize(options.kmsProviders);
-      } else if (!options.onKmsProviderRefresh) {
+      } else if (!options.onKmsProviderRefresh && !options.onEmptyKmsProviders) {
         throw new TypeError('Need to specify either kmsProviders ahead of time or when requested');
       }
 
       this._onKmsProviderRefresh = options.onKmsProviderRefresh;
+      this._onEmptyKmsProviders = options.onEmptyKmsProviders;
```

Comment:
What’s the difference between these two?

Reply:
`onKmsProviderRefresh` is the user supplied function to populate the KMS providers on demand if they want. `onEmptyKmsProviders` is what the driver will set as the last fallback to pull credentials from the environment if 1) They didn't provide them to the mongo client and 2) the user provided on demand callback was not provided or returned empty results.

Reply:
Ah that's a really good idea - so much better than this.